src/Sulu/Bundle/HttpCacheBundle/CacheLifetime/CacheLifetimeEnhancer.php

```diff
@@ -33,14 +33,21 @@ class CacheLifetimeEnhancer implements CacheLifetimeEnhancerInterface
      */
     private $sharedMaxAge;
 
+    /**
+     * @var CacheLifetimeRequestEnhancer
+     */
+    private $cacheLifetimeRequestEnhancer;
+
     public function __construct(
         CacheLifetimeResolverInterface $cacheLifetimeResolver,
         $maxAge,
-        $sharedMaxAge
+        $sharedMaxAge,
+        CacheLifetimeRequestEnhancer $cacheLifetimeRequestEnhancer
```

Comment:
For bc this should be:

```suggestion
       ?CacheLifetimeRequestEnhancer $cacheLifetimeRequestEnhancer = null
```

and throw a deprecation message when not set.